pages/episodes/[currentEpisode].js

```diff
@@ -1,37 +1,107 @@
+export default function EpisodePage() {
+  const [allEpisodes] = useAtom(episodes);
 
   const router = useRouter();
   const { currentEpisode } = router.query;
 
-  if (error) return <div>failed to load</div>;
-  if (!currentEpisode || isLoading) return <div>loading...</div>;
-
-  if (data) {
-    const { serie } = data;
-
-    const [filteredEpisode] = serie.filter(
-      (episode) => episode.nummer === parseInt(currentEpisode)
-    );
-
-    if (!filteredEpisode) {
-      return (
-        <main>
-          <h2>Hoppla, diese Folge scheint es nicht zu geben.</h2>
-        </main>
-      );
-    }
+  const [filteredEpisode] = allEpisodes.filter(
+    (episode) => episode.nummer === parseInt(currentEpisode)
+  );
 
+  if (!filteredEpisode) {
     return (
       <main>
-        <EpisodeDetails episode={filteredEpisode} />
+        <h2>Diese Folge scheint es nicht zu geben.</h2>
       </main>
     );
   }
+
+  const {
+    nummer: number,
+    titel: title,
+    autor: author,
+    hörspielskriptautor: scriptauthor,
+    beschreibung: description,
+    veröffentlichungsdatum: releasedate,
+    kapitel: chapters,
+    links,
+    teile: parts,
+  } = filteredEpisode;
+
+  return (
+    <main>
+      <EpisodeCard>
+        <EpisodeCardHeader episodeNumber={number} />
+        {getCoverURL(links) ? (
+          <EpisodeImage
+            src={getCoverURL(links)}
+            alt={`Folge ${number}, Die Drei Fragezeichen ${title}`}
+            width={500}
+            height={500}
+            priority
+          />
+        ) : (
+          <NoContentMessage>kein Artwork vorhanden</NoContentMessage>
+        )}
+        {author ? (
+          <EpisodeFacts>Autor: {author}</EpisodeFacts>
+        ) : (
+          <NoContentMessage>
+            {parts ? "" : "kein Autor angegeben"}
+          </NoContentMessage>
+        )}
+        {scriptauthor ? (
+          <EpisodeFacts>Hörspielskript-Autor: {scriptauthor}</EpisodeFacts>
+        ) : (
+          <NoContentMessage>
+            kein Hörspielskript-Autor angegeben
+          </NoContentMessage>
+        )}
+        {releasedate ? (
+          <EpisodeFacts>
+            Veröffentlichungsdatum: {getFormattedDate(releasedate)}
+          </EpisodeFacts>
+        ) : (
+          <NoContentMessage>
+            kein Veröffentlichungsdatum angegeben
+          </NoContentMessage>
+        )}
+        {description ? (
+          <EpisodeDescription description={description} />
+        ) : (
+          <NoContentMessage>keine Beschreibung vorhanden</NoContentMessage>
+        )}
+        {chapters?.length > 0 ? (
+          <Chapters chapters={chapters} />
+        ) : (
+          <NoContentMessage>
+            {parts ? "" : "keine Kapitelliste vorhanden"}
+          </NoContentMessage>
+        )}
+        {parts?.length > 0
+          ? parts.map((part) => {
+              return <Parts key={part.buchstabe} part={part} />;
+            })
+          : ""}
+      </EpisodeCard>
```

Comment:
In general I suggest writing your components so they are complete units, ready to use. In your components/episode folder you have a number of singular components that do not make a lot of sense when used on their own.
This is not a must change for you right now, but I recommend writing component like this:
```
<main>
   <EpisodeCard
      number={nummer}
      description={beschreibung}
      ...
      />
</main>
```

Reply:
Hi Thomas, thanks for the recommendation. 
I actually had all of my episode components stacked up in the `EpisodeCard` before, but this would mean passing a lot of logic and props into and through this component. 
See [previous pull request](https://github.com/RobinWitt/capstone-project/blob/feature/favorite-page/components/EpisodeDetails/index.js) for comparison.
Maybe I can find a compromise like moving the bottom part back to the `EpisodeCard` again but keeping the `EpisodeHeader` directly on the page. I'll think about it.